File: Frontend/sakai-ng-master/src/app/biblioteca/interfaces/notificacion.ts
export class Notificacion {
  id: number;
  usuarioDestino: string;
  mensaje: string;
  fechaCreacion: string;  // o Date según tu JSON
  leida: boolean;

  constructor(init?: Partial<Notificacion>) {
    this.id = 0;
    this.usuarioDestino = '';
    this.mensaje = '';
    this.fechaCreacion = '';
    this.leida = false;
    Object.assign(this, init);
  }
}

File: Backend/login-microsoft365/src/main/java/com/miapp/model/Notificacion.java
package com.miapp.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data            // genera getters, setters, toString, equals, hashCode…
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "NOTIFICACION")
@SequenceGenerator(
        name="seq_notificacion",
        sequenceName="SEQ_NOTIFICACION",  // o el nombre de tu sequence en Oracle
        allocationSize=1
)
public class Notificacion {
    @Id
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "seq_notificacion"
    )
    @Column(name="ID")
    private Long id;

    @Column(name="USUARIO")
    private String usuarioDestino;

    private String mensaje;

    @Column(name = "FECHA_CREACION")
    private LocalDateTime fechaCreacion;

    // si usas NUMBER(1):
    @Column(name="LEIDO")
    private Boolean leida;
}


